metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlTable.java

```diff
@@ -169,56 +176,93 @@ public boolean exists(final QualifiedName name) {
+     *  1. Gets the table parameters and locks the requested records. If lock cannot be attained,
+     *  the request to update fails
+     *  2. Validates the metadata location
+     *  3. If validated, updates the table parameters.
+     * @param tableInfo table info
      */
     @Transactional(propagation = Propagation.REQUIRES_NEW)
-    public void lockIcebergTable(final Long tableId, final QualifiedName tableName) {
-        String tableType = null;
+    public void updateIcebergTable(final TableInfo tableInfo) {
+        final QualifiedName tableName = tableInfo.getName();
+        final Long tableId = getTableId(tableName);
+        Map<String, String> existingTableMetadata = Maps.newHashMap();
+        log.debug("Lock Iceberg table {}", tableName);
         try {
-            tableType = jdbcTemplate.queryForObject(SQL.TABLE_PARAM_LOCK,
-                new SqlParameterValue[] {new SqlParameterValue(Types.BIGINT, tableId),
-                    new SqlParameterValue(Types.VARCHAR, PARAM_TABLE_TYPE), }, String.class);
-        } catch (EmptyResultDataAccessException ignored) { }
-        if (tableType == null || !ICEBERG_TABLE_TYPE.equalsIgnoreCase(tableType)) {
-            final String message = String.format("Originally table %s is not of type iceberg", tableName);
-            log.info(message);
+            existingTableMetadata = jdbcTemplate.query(SQL.TABLE_PARAMS_LOCK,
+                new SqlParameterValue[]{new SqlParameterValue(Types.BIGINT, tableId)}, rs -> {
+                    final Map<String, String> result = Maps.newHashMap();
+                    while (rs.next()) {
+                        result.put(rs.getString(COL_PARAM_KEY), rs.getString(COL_PARAM_VALUE));
+                    }
+                    return result;
+                });
+        } catch (EmptyResultDataAccessException ex) {
+            log.info(String.format("No parameters defined for iceberg table %s", tableName));
+        } catch (Exception ex) {
+            final String message = String.format("Failed getting a lock on iceberg table %s", tableName);
+            log.warn(message, ex);
             throw new InvalidMetaException(tableName, message, null);
         }
-        Boolean isLocked = null;
-        try {
-            isLocked = jdbcTemplate.queryForObject(SQL.TABLE_PARAM_LOCK,
-                new SqlParameterValue[] {new SqlParameterValue(Types.BIGINT, tableId),
-                    new SqlParameterValue(Types.VARCHAR, PARAM_METADATA_LOCK), }, Boolean.class);
-        } catch (EmptyResultDataAccessException ignored) { }
-        if (isLocked == null) {
-            insertTableParam(tableId, PARAM_METADATA_LOCK, 1);
-        } else if (isLocked) {
-            final String message = String.format("Iceberg table %s is locked.", tableName);
+        if (existingTableMetadata.isEmpty()
+            || !ICEBERG_TABLE_TYPE.equalsIgnoreCase(existingTableMetadata.get(PARAM_TABLE_TYPE))) {
```

Comment:
if (existingTableMetadata.get(PARAM_METADATA_LOCATION) ) check move here as well. It's better to have a function to check/valide the existingTableMetadata .

Reply:
I will move this check to the validateIcebergUpdate method.